firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -884,4 +884,39 @@ public void testCollectionGroupQueriesWithWhereFiltersOnArbitraryDocumentIds() {
                 .get());
     assertEquals(asList("cg-doc2"), querySnapshotToIds(querySnapshot));
   }
+
+  // See: https://github.com/firebase/firebase-android-sdk/issues/3528
+  @Test
+  public void testAddThenUpdatesWhileOffline() {
```

Comment:
Ideally, these would be LocalStoreTests.

Reply:
Added TODO.